packages/toolpad-studio-components/src/DataGrid.tsx

```diff
@@ -479,11 +479,11 @@ function getNarrowedColType(type?: string): GridColType | undefined {
   return (type && type in DEFAULT_COLUMN_TYPES ? type : undefined) as GridColType | undefined;
 }
 
-export function parseColumns(columns: SerializableGridColumns): GridColDef[] {
+export function parseColumns(columns: SerializableGridColumns, isEditable?: boolean): GridColDef[] {
```

Comment:
`parseColumns` acts as a conversion between "serializable columns" and "X grid columns". It's being reused across the codebase for this purpose.
Let's instead solve this downstream. Where we iterate a second time to create the rendered coluns:
https://github.com/mui/mui-toolpad/blob/02cc9a97b76098d51651eb48cf10c12ccdc7c8fe/packages/toolpad-studio-components/src/DataGrid.tsx#L1264-L1279

Reply:
@Janpot , say we're having `id` column it shouldn't be editable at all right, should I move the `isIdColumn` logic to this function? 

If we update the `editable` here `id` column may probably get override right?

Reply:
How about only setting `isEditable` if it isn't already set? And remove `isEditable: true` from the `baseColumn` in `parseColumns`.

Reply:
Yea, that's a good approach. Will update accordingly.